icons: Add matchTextDirection to directional icons

yash-agarwa-l opened the pull request:
This enables automatic mirroring for directional icons (like arrows and send buttons) in RTL languages.

Previously, icons like `chevron_right` and `send` would always point to the right, even in RTL layouts where the flow of the application is right-to-left. This updates the icon generation script to set `matchTextDirection: true` for these specific icons, ensuring they flip correctly.

This addresses the fourth RTL issue identified by @gnprice in [this comment](https://github.com/zulip/zulip-flutter/pull/1991#issuecomment-3573671163), and fixes #1907.

### Changes
* Updated `tools/icons/build-icon-font.js` to add `matchTextDirection: true` for `arrow_right`, `chevron_right`, and `send`.
* Updated `lib/widgets/icons.dart` with the regenerated code.

Send Button
<table> <thead> <tr> <th>State</th> <th align="center">LTR </th> <th align="center">RTL </th> </tr> </thead> <tbody> <tr> <td align="center"><strong>Before Send</strong></td> <td align="center"> <img src="https://github.com/user-attachments/assets/efc978da-95fa-4ee9-80e6-403618d254b8" alt="Before LTR Send Button" width="300"/> </td> <td align="center"> <img src="https://github.com/user-attachments/assets/eec472e6-7943-4fef-9710-e7327a3d8087" alt="Before RTL Send Button" width="300"/> </td> </tr> <tr> <td align="center"><strong>After Send</strong></td> <td align="center"> <!-- LTR stays the same before/after as requested --> <img src="https://github.com/user-attachments/assets/efc978da-95fa-4ee9-80e6-403618d254b8" alt="After LTR Send Button (unchanged)" width="300"/> </td> <td align="center"> <img src="https://github.com/user-attachments/assets/c66b8671-ce5f-47de-9850-d79a7fdd1a13" alt="After RTL Send Button" width="300"/> </td> </tr> </tbody> </table>
Thumbs Up Button (Small Arrow)
<table> <thead> <tr> <th>State</th> <th align="center">LTR </th> <th align="center">RTL</th> </tr> </thead> <tbody> <tr> <td align="center"><strong>Before ` > ` </strong></td> <td align="center"> <img src="https://github.com/user-attachments/assets/4ac13376-a46c-46d2-b78f-811255bfa394" alt="Before LTR ` >` " width="300"/> </td> <td align="center"> <img src="https://github.com/user-attachments/assets/ed426302-e5ef-4412-8ef8-e0a017ae657a" alt="Before RTL `>` " width="300"/> </td> </tr> <tr> <td align="center"><strong>` > `</strong></td> <td align="center"> <!--  --> <img src="https://github.com/user-attachments/assets/4ac13376-a46c-46d2-b78f-811255bfa394" alt="After LTR  ` >` (unchanged)" width="300"/> </td> <td align="center"> <!-- Using the 'better' version for after state --> <img src="https://github.com/user-attachments/assets/a072e708-bf9d-4486-b7ae-46111095907e" alt="After RTL Thumbs Up Button" width="300"/> </td> </tr> </tbody> </table>
More Button (…)
<table> <thead> <tr> <th>State</th> <th align="center">LTR </th> <th align="center">RTL </th> </tr> </thead> <tbody> <tr> <td align="center"><strong>Before More</strong></td> <td align="center"> <!--  --> <img src="https://github.com/user-attachments/assets/20aea04d-81fa-4fc4-a05b-e51573ee98a5" alt="Before LTR More Button" width="300"/> </td> <td align="center"> <!-- Approximated using pre-change RTL layout --> <img src="https://github.com/user-attachments/assets/517dc94e-fe52-48f1-86ff-9fba9414478a" alt="Before RTL More Button" width="300"/> </td> </tr> <tr> <td align="center"><strong>After More</strong></td> <td align="center"> <!-- LTR stays the same before/after as requested --> <img src="https://github.com/user-attachments/assets/20aea04d-81fa-4fc4-a05b-e51573ee98a5" alt="After LTR More Button (unchanged)" width="300"/> </td> <td align="center"> <img src="https://github.com/user-attachments/assets/6eef1347-e7b1-4f19-a1b3-e29157648a82" alt="After RTL More Button" width="300"/> </td> </tr> </tbody> </table>

chrisbobbe commented:
(Oops, I didn't mean to close this—accidentally clicked the wrong button.)

chrisbobbe commented:
Ah, and also: there are some other icons with arrows that should probably get the same treatment:

- `arrow_left_right`
- (Hmm, maybe just that one—I'd thought there was another, but I guess not.)

And others that aren't arrows but also look like they should flip because they represent text in a directional layout:

- `message_feed`
- `topics`

yash-agarwa-l commented:
Thanks for the refactor commits, @chrisbobbe! I've pulled them in and squashed everything into a single commit.

I also updated directional_icons.json to include message_feed, topics, and arrow_left_right.

I am a bit unsure about arrow_left_right, though—since it depicts a concept (two-way exchange) rather than a specific UI direction, I wonder if it usually stays static? I’ve included it for now to be safe, but I'm happy to remove it if you prefer.

gnprice commented:
Thanks! Looks good; merging, with two further tweaks. First:

         We update the icon generation script to automatically set
    -    matchTextDirection: true for icons that imply directionality
    -    (arrow_right, chevron_right, send).
    -
    +    matchTextDirection: true for icons that imply directionality.
         This ensures these icons flip correctly in RTL locales.

This had a list of icons affected which wasn't accurate. (It looks like it was from a previous version of the commit.)

Could correct the list — or just cut it, because after all that information is plenty clear in the actual diff.

Second, added the following commit on top, taking advantage of that nice comment-able file format as discussed above:
1c609cc12 icons [nfc]: Expand comments on list of directional icons
